packages/web-components/src/avatar/avatar.ts

```diff
@@ -0,0 +1,175 @@
+import { attr, FASTElement, nullableNumberConverter } from '@microsoft/fast-element';
+import { getInitials } from '../utils/get-initials.js';
+import {
+  AvatarActive,
+  AvatarAppearance,
+  AvatarColor,
+  AvatarNamedColor,
+  AvatarShape,
+  AvatarSize,
+} from './avatar.options.js';
+
+/**
+ * The base class used for constructing a fluent-avatar custom element
+ * @public
+ */
+export class Avatar extends FASTElement {
+  /**
+   * The name of the person or entity represented by this Avatar. This should always be provided if it is available.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr
+  public name: string | undefined;
+  protected nameChanged(previous: string, next: string): void {
+    if (next !== undefined && !this.initials) {
+      this.setInitials();
+    }
+  }
```

Comment:
What callback? J/K - Good catch it all executes in the template :). 

The one which I've left here is `color-id` because there's a scenario where colorful is already set and a change to `color-id` doesn't kick anything off in the template.

Reply:
`color-id` change still re-renders the `data-color` part of the template, see https://github.com/microsoft/fluentui/pull/26729#discussion_r1114042563